cpp/ql/lib/experimental/semmle/code/cpp/dataflow/ProductFlow.qll

```diff
@@ -0,0 +1,119 @@
+import semmle.code.cpp.ir.dataflow.DataFlow
+import semmle.code.cpp.ir.dataflow.DataFlow2
+
+module ProductFlow {
+  abstract class Configuration extends string {
```

Comment:
May as well make this a parameterized module.

Reply:
Can we do that yet? When I try the straightforward thing I get an error: `Classes may not extend across parameterised module boundaries`.

Reply:
Yep, that's still a restriction that's yet to be lifted.

Reply:
Is it even going to be lifted? I think extending an abstract class within a parameterized module will have weird implications for the "active module" concept. I'm guessing this will need to wait until the dataflow API is a parameterized module.